### PR TITLE
For #5884: Adds user-specified search engines telemetry

### PR DESCRIPTION
**Pull request by ValentinTimisica**
Created events for adding and deleting a custom search engine and for searches
performed with a custom search engine.

---

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks.
- [x] **Tests**: This PR does not include thorough tests. Telemetry task.
- [x] **Screenshots**: This PR does not include screenshots or GIFs of the changes made.
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md).

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by ValentinTimisica** [comment]
# Request for data collection review form

**All questions are mandatory. You must receive review from a data steward peer on your responses to these questions before shipping new data collection.**

1) **What questions will you answer with this data?**

- How often user add or use other search engines not listed?

2) **Why does Mozilla need to answer these questions?  Are there benefits for users? Do we need this information to address product or business requirements?**

- Determine how to better integrate frequently used features within the product.

3) **What alternative methods did you consider to answer these questions? Why were they not sufficient?**

- N/A (These are baseline metrics)

4) **Can current instrumentation answer these questions?**
- No, adding custom search engines is a new feature with no existing telemetry

5) **List all proposed measurements and indicate the category of data collection for each measurement, using the Firefox [data c](https://wiki.mozilla.org/Firefox/Data_Collection)[ollection ](https://wiki.mozilla.org/Firefox/Data_Collection)[categories](https://wiki.mozilla.org/Firefox/Data_Collection) on the found on the Mozilla wiki.**

- All data is Category 2.

6) **How long will this data be collected?**

- Until 03/01/2020

7) **What populations will you measure?**

- All release, beta, and nightly users with telemetry enabled.

8) **Please provide a general description of how you will analyze this data.**

- Glean / Amplitude

9) **Where do you intend to share the results of your analysis?**

- Only on Glean, Amplitude and with mobile teams.

**Comment by ValentinTimisica** [comment]
I put the expiration date for this telemetry to be 03/01/2020 because this is the same date used for all the other telemetry probes. Not sure though this telemetry will be stopped at that date.

File: app/src/main/java/org/mozilla/fenix/search/SearchController.kt [hunk 1]
```diff
@@ -58,6 +58,13 @@ class DefaultSearchController(
             }
 
             context.metrics.track(event)
+            if (CustomSearchEngineStore.isCustomSearchEngine(
+                    context,
+                    store.state.searchEngineSource.searchEngine.identifier
+                )
+            ) {
+                context.components.analytics.metrics.track(Event.SearchWithCustomEngine)
```
Comment (liuche) [inline app/src/main/java/org/mozilla/fenix/search/SearchController.kt]: How is this probe different form the event in `createSearchEvent` above? It looks like it also sends whether or not a search is custom or not https://github.com/mozilla-mobile/fenix/blob/master/app/src/main/java/org/mozilla/fenix/search/SearchController.kt#L130

Comment (ValentinTimisica) [reply app/src/main/java/org/mozilla/fenix/search/SearchController.kt]: From what I understand, `createSearchEvent` creates a `Event.PerformedSearch` which records:
 - a [search event](https://github.com/mozilla-mobile/fenix/blob/8aa04c048e9df949b0be31bbc9c079fb57be2c77/app/src/main/java/org/mozilla/fenix/components/metrics/GleanMetricsService.kt#L103) that can have only [these](https://github.com/mozilla-mobile/fenix/blob/8aa04c048e9df949b0be31bbc9c079fb57be2c77/app/metrics.yaml#L62) values (no 'custom' here).
- a [counter](https://github.com/mozilla-mobile/fenix/blob/8aa04c048e9df949b0be31bbc9c079fb57be2c77/app/src/main/java/org/mozilla/fenix/components/metrics/GleanMetricsService.kt#L102) with the labels `<search-engine-name>.<source>` (Description for `search_count` [here](https://github.com/mozilla-mobile/fenix/blob/8aa04c048e9df949b0be31bbc9c079fb57be2c77/app/metrics.yaml#L322)). So this counter is the place where "custom" is used. 

@boek is this correct?

Is this counter enough for what we need here? 
If so I will remove my event `Event.SearchWithCustomEngine`(this event only tracks if the user performed a search with a custom search engine).
How should i proceed?

Comment (boek) [reply app/src/main/java/org/mozilla/fenix/search/SearchController.kt]: Yes this is correct. We can probably remove `SearchWithCustomEngine` :)

**Pull request merged**